frontend/packages/shared/src/components/TreeView/utils/domUtils.test.tsx

```diff
@@ -0,0 +1,513 @@
+      renderDom();
+      expect(findNextVisibleNodeId(rootId, thirdItemId)).toBeNull();
+    });
+  });
+
+  describe('findPreviousVisibleNodeId', () => {
+    const rootId = 'rootId';
+    const firstItemId = 'firstItemId';
+    const secondItemId = 'secondItemId';
+    const subItemId = 'subItemId';
+    const invisibleSubSubItemId = 'subSubItemId';
+    const thirdItemId = 'thirdItemId';
+    const renderDom = () =>
+      render(
+        <ul role='tree' id={rootId}>
+          {renderListItem(rootId, firstItemId, 'firstItem', 1)}
+          {renderListItem(rootId, secondItemId, 'secondItem', 1, true, [
+            renderListItem(rootId, subItemId, 'subItem', 2, false, [
+              renderListItem(rootId, invisibleSubSubItemId, 'subSubItem', 3),
+            ]),
+          ])}
+          {renderListItem(rootId, thirdItemId, 'thirdItem', 1)}
+        </ul>,
+      );
+
+    it('Returns the id of the previous visible node', () => {
+      renderDom();
+      expect(findPreviousVisibleNodeId(rootId, secondItemId)).toBe(firstItemId);
+      expect(findPreviousVisibleNodeId(rootId, subItemId)).toBe(secondItemId);
+      expect(findPreviousVisibleNodeId(rootId, thirdItemId)).toBe(subItemId);
+    });
+
+    it('Returns null if the given node is the first visible node', () => {
+      renderDom();
+      expect(findPreviousVisibleNodeId(rootId, firstItemId)).toBeNull();
+    });
+  });
+
+  describe('findNodeIndexWithinGroup', () => {
+    const rootId = 'rootId';
+    const firstItemId = 'firstItemId';
+    const secondItemId = 'secondItemId';
+    const subItem2AId = 'subItem2AId';
+    const subItem2BId = 'subItem2BId';
+    const thirdItemId = 'thirdItemId';
+    const renderDom = () =>
+      render(
+        <ul role='tree' id={rootId}>
+          {renderListItem(rootId, firstItemId, 'firstItem', 1)}
+          {renderListItem(rootId, secondItemId, 'secondItem', 1, true, [
+            renderListItem(rootId, subItem2AId, 'subItem2A', 2),
+            renderListItem(rootId, subItem2BId, 'subItem2B', 2),
+          ])}
+          {renderListItem(rootId, thirdItemId, 'thirdItem', 1)}
+        </ul>,
+      );
+
+    it('Returns the index of the node within its group', () => {
+      renderDom();
+      expect(findNodeIndexWithinGroup(rootId, firstItemId)).toBe(0);
+      expect(findNodeIndexWithinGroup(rootId, secondItemId)).toBe(1);
+      expect(findNodeIndexWithinGroup(rootId, thirdItemId)).toBe(2);
+      expect(findNodeIndexWithinGroup(rootId, subItem2AId)).toBe(0);
+      expect(findNodeIndexWithinGroup(rootId, subItem2BId)).toBe(1);
+    });
+
+    it('Returns the correct index when the given node is a root node', () => {
+      renderDom();
+      expect(findNodeIndexWithinGroup(rootId, firstItemId)).toBe(0);
+      expect(findNodeIndexWithinGroup(rootId, secondItemId)).toBe(1);
+      expect(findNodeIndexWithinGroup(rootId, thirdItemId)).toBe(2);
+    });
+  });
```

Comment:
Are the three first tests in the first it-test the same as the tests in the second it-test? 🤔

And should there be a test doing this? If it could happen at this point?
`findNodeIndexWithinGroup(rootId, rootId)`

Reply:
> Are the three first tests in the first it-test the same as the tests in the second it-test? 🤔

Good catch! I must have decided to split the first test in two, but forgotten to remove the checks.

> And should there be a test doing this? If it could happen at this point?
`findNodeIndexWithinGroup(rootId, rootId)`

I don't think that's relevant, it's not being called from the root component.